webapp/src/ddocs/medic-client/views/messages_by_contact_date/map.js

```diff
@@ -1,39 +1,22 @@
 function(doc) {
-
-  var getLineage = function(contact) {
-    var lineage = [];
-    while (contact) {
-      if (contact._id) {
-        lineage.push(contact._id);
-      }
-      contact = contact.parent;
-    }
-    return lineage;
-  };
-
-  var emitMessage = function(doc, contact, phone, message) {
-    var from = (contact && contact._id) || phone;
-    var id = from || doc._id;
+  var emitMessage = function(doc, contact, phone) {
+    var id = (contact && contact._id) || phone || doc._id;
     emit([ id, doc.reported_date ], {
       id: doc._id,
```

Comment:
I'm pretty sure we don't need to include the `doc._id` in the value - the ID is always returned anyway, so this is duplicated.